src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/mapper/induction/InductionEntityMapper.kt

```diff
@@ -78,31 +78,61 @@ abstract class InductionEntityMapper {
+  fun updateFutureWorkInterests(entity: InductionEntity, dto: UpdateInductionDto): FutureWorkInterestsEntity? {
+    return if (entity.futureWorkInterests == null) {
+      futureWorkInterestsEntityMapper.fromUpdateDtoToEntity(dto.futureWorkInterests)
+    } else {
+      futureWorkInterestsEntityMapper.updateEntityFromDto(entity.futureWorkInterests, dto.futureWorkInterests)
+        .let { entity.futureWorkInterests }
+    }
+  }
+
+  fun updateInPrisonInterests(entity: InductionEntity, dto: UpdateInductionDto): InPrisonInterestsEntity? {
+    return if (entity.inPrisonInterests == null) {
+      inPrisonInterestsEntityMapper.fromUpdateDtoToEntity(dto.inPrisonInterests)
+    } else {
+      inPrisonInterestsEntityMapper.updateEntityFromDto(entity.inPrisonInterests, dto.inPrisonInterests)
+        .let { entity.inPrisonInterests }
+    }
+  }
+
+  fun updatePersonalSkillsAndInterests(entity: InductionEntity, dto: UpdateInductionDto): PersonalSkillsAndInterestsEntity? {
+    return if (entity.personalSkillsAndInterests == null) {
+      skillsAndInterestsEntityMapper.fromUpdateDtoToEntity(dto.personalSkillsAndInterests)
+    } else {
+      skillsAndInterestsEntityMapper.updateEntityFromDto(entity.personalSkillsAndInterests, dto.personalSkillsAndInterests)
+        .let { entity.personalSkillsAndInterests }
+    }
+  }
+
+  fun updatePreviousQualifications(entity: InductionEntity, dto: UpdateInductionDto): PreviousQualificationsEntity? {
+    return if (entity.previousQualifications == null) {
+      previousQualificationsEntityMapper.fromUpdateDtoToEntity(dto.previousQualifications)
+    } else {
+      previousQualificationsEntityMapper.updateEntityFromDto(entity.previousQualifications, dto.previousQualifications)
+        .let { entity.previousQualifications }
+    }
+  }
+
+  fun updatePreviousTraining(entity: InductionEntity, dto: UpdateInductionDto): PreviousTrainingEntity? {
+    return if (entity.previousTraining == null) {
+      previousTrainingEntityMapper.fromUpdateDtoToEntity(dto.previousTraining)
+    } else {
+      previousTrainingEntityMapper.updateEntityFromDto(entity.previousTraining, dto.previousTraining)
+        .let { entity.previousTraining }
+    }
+  }
+
+  fun updatePreviousWorkExperiences(entity: InductionEntity, dto: UpdateInductionDto): PreviousWorkExperiencesEntity? {
+    return if (entity.previousWorkExperiences == null) {
+      workExperiencesEntityMapper.fromUpdateDtoToEntity(dto.previousWorkExperiences)
+    } else {
+      workExperiencesEntityMapper.updateEntityFromDto(entity.previousWorkExperiences, dto.previousWorkExperiences)
+        .let { entity.previousWorkExperiences }
+    }
+  }
 
   fun updateWorkOnRelease(entity: InductionEntity, dto: UpdateInductionDto): WorkOnReleaseEntity? =
-    workOnReleaseEntityMapper.updateEntityFromDto(entity.workOnRelease, dto.workOnRelease)
+    workOnReleaseEntityMapper.updateEntityFromDto(entity.workOnRelease!!, dto.workOnRelease)
```

Comment:
`workOnRelease` can never be null.

Reply:
Would changing the method signature of `workOnReleaseEntityMapper.updateEntityFromDto` be a better approach here? Or can it be null in _some_ cases, just not here?

Reply:
The entity fields are always nullable, which is something I'd like to investigate at some point.